packages/protocol/lib/web3-utils.ts

```diff
@@ -387,6 +393,28 @@ export function getFunctionSelectorsForContract(contract: any, contractName: str
   return selectors
 }
 
+export function checkInheritance(baseContractName: string, derivativeContractArtifact: any) {
+  return derivativeContractArtifact.ast.nodes.find(
+    (astNode: any) => astNode.nodeType === 'ImportDirective' && (astNode.file as string).endsWith(`${baseContractName}.sol`)
+  ) !== undefined
+}
+
+export function checkInheritanceDeep(baseContractName: string, derivativeContractArtifact: any, artifacts: any) {
+  const isImport = (astNode: any) => astNode.nodeType === 'ImportDirective'
+  const imports: any[] = derivativeContractArtifact.ast.nodes.filter((astNode: any) => isImport(astNode))
+  while (imports.length) { // BFS 
```

Comment:
Big nit, but looks more like a DFS to me ;)

Reply:
Haha great catch. I thought I was using a queue but its actually a stack 🤦‍♂️